tests/src/core/testqgstranslateproject.cpp

```diff
@@ -52,16 +52,12 @@ class TestQgsTranslateProject : public QObject
 void TestQgsTranslateProject::initTestCase()
 {
   //start application
-  QgsApplication::init();
```

Comment:
This is done in a lot of places, mostly in tests which require a custom QSettings environment.

What do you think about modifying the QGSTEST_MAIN macro so that we always use a custom QSettings environment for each test, automatically cleared before the test is executed? 

I can't see any issue if we did this, and the benefit is that it would nicely isolate local profile settings away from the tests (tests won't be affected by local settings, and local setting profiles won't be changed by the test).

Reply:
QGSTEST_MAIN should already do that after this PR on a per test basis. Here the issue is about a per test case basis, not something that can be fixed in QGSTEST_MAIN AFAICT. And I wouldn't be surprised if we had tests that rely on the current behavior. I'll just try to fixup the failing ones.